plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java

```diff
@@ -166,6 +167,7 @@ private void writeBlock(BlockBuilder blockBuilder, Type columnType, int columnId
             writeBooleanBlock(blockBuilder, columnType, columnIdx);
         }
         else if (javaType.equals(long.class)) {
+            // Applies to timestamp as well
```

Comment:
only for ShortTimestampType i.e. precision <= 6.

Reply:
I'll make an additional note